### PR TITLE
SDK dogfooding

### PR DESCRIPTION
**Pull request by shoom3301**
SDK refactoring: https://github.com/cowprotocol/cow-sdk/pull/97

# Changes
1. Removed `optimize-bundle` script, we don't need it since we got rid of redundant dependencies
2. Most of the changes are related to `SupportedChainId`, `OrderKind`, `OrderClass` imports
3. Added refactored version of `metadataApiSDK` usage
4. Integrated refactored version of `OrderBookApi`
5. Changed imports for types related to the order-book
6. Removed irrelevant `Gas price strategy` code
7. Removed `0x` and `Matcha` APIs due to their irrelevance
8. Replaced `getNativePrice` by import from SDK
9. Integrated `sendOrder` and `sendSignedOrderCancellation` form SDK
10. Added lazy loading for signing utils from `@cowprotocol/contracts`, because they are pretty heavy and we don't need them most of the time

Other PRs: https://github.com/cowprotocol/cowswap/pulls?q=is%3Apr+is%3Aclosed+label%3ASDK_dogfooding

### CODE REVIEW

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , great job!
But I have found 1 issue that blocks my approval: AppData details is not working in your PR
image
image

File: src/apps/explorer/components/SummaryCardsWidget/VolumeChart/useGetVolumeData.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export function useGetVolumeData(volumeTimePeriod = VolumePeriod.DAILY): VolumeD
 }
 
 async function getLastHoursData(network: Network): Promise<RawVolumeItem[]> {
-  const data = await COW_SDK.cowSubgraphApi.getLastHoursVolume(48, { chainId: network })
+  const data = await subgraphApiSDK.getLastHoursVolume(48, { chainId: network })
```
Comment (alfetopito) [inline src/apps/explorer/components/SummaryCardsWidget/VolumeChart/useGetVolumeData.ts:42]: One thing I just realized as I changed this yesterday: Subgraph also has staging endpoints.
True, it's configured per environment with env vars, and maybe we don't need to change anything here.
Just for you to be aware

File: src/const.ts [hunk 2]
```diff
@@ -160,31 +158,6 @@ export const ORDER_BOOK_HOPS_MAX = 30
...
-  if (goerliUrl) {
-    subgraphBaseUrls[ChainId.GOERLI] = goerliUrl
-  }
-
-  return subgraphBaseUrls
-}
-
-export const COW_SDK = new CowSdk(Network.MAINNET, {
-  subgraphBaseUrls: getSubgraphUrls(),
-})
```
Comment (alfetopito) [inline src/const.ts:186]: Is this logic moving somewhere else?
Otherwise, how will the subgraph urls be passed onto the subgraph instance?

**Review by alfetopito — COMMENTED** [review]
I'm concerned with the subgraph env var urls.

See the comment

File: src/apps/explorer/components/SummaryCardsWidget/VolumeChart/useGetVolumeData.ts (hunk 1 above)
Comment (shoom3301) [reply src/apps/explorer/components/SummaryCardsWidget/VolumeChart/useGetVolumeData.ts:42]: Thanks! Checked staging and prod - they use the same endpoints

File: src/const.ts (hunk 2 above)
Comment (shoom3301) [reply src/const.ts:186]: Thank you!
Added it:
https://github.com/cowprotocol/explorer/pull/391/commits/005c1c39d5cd6d1643abbe2233434a6d80a72887

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed

**Review by elena-zh — APPROVED** [review]
Hey @shoom3301 , I'm approving the current PR. However, I found a list of new issues I faced for the updated AppData page, and reported them  in #392. It would be great to fix them.

**Review by alfetopito — APPROVED** [review]
👍

**Pull request merged**